src/main/webapp/creatives/maps/simid_map_creative.test.js

```diff
@@ -63,3 +72,32 @@ test('testing button text updates with default ad params', () => {
     expect(button.innerText).toBe('Find Nearest Location');
 });
 
+test('instance of map is instantiated on button click', () => {
+    const eventData = {
+        args: {
+            creativeData: {
+                adParameters: '{}',
+            },
+            environmentData: {},
+        },
+        messageId: 0,
+        sessionId: "test-session-id",
+        timestamp: 0,
+        type: "SIMID:Player:init",
+    };
+    testMap.onInit(eventData);
+
+    const startData = {
+        messageId: 1,
+        sessionId: "test-session-id",
+        timestamp: 0,
+        type: "SIMID:Player:startCreative",  
+    }
+    testMap.onStart(startData);
+
+    const findNearestButton = document.getElementById('findNearest');
+    const mapDiv = document.getElementById("map");
+    findNearestButton.dispatchEvent(new Event('click'));
+    expect(window.google.maps.Map.mock.instances.length).toBe(1);
```

Comment:
Do you want to test that Maker gets instantiated as well?
Maybe check that the co-ordinates are passed through also

Reply:
Good call, fixed